lib/services/plugins-service.ts

```diff
@@ -0,0 +1,196 @@
+				}
+				
+				if(!skipExecution) {
+					this.$fs.ensureDirectoryExists(pluginDestinationPath).wait();
+					shelljs.cp("-R", pluginData.fullPath, pluginDestinationPath);
+				
+					let pluginPlatformsFolderPath = path.join(pluginDestinationPath, pluginData.name, "platforms", platform);
+					let pluginConfigurationFilePath = path.join(pluginPlatformsFolderPath, platformData.configurationFileName);					
+					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						let pluginConfigurationFileContent = this.$fs.readText(pluginConfigurationFilePath).wait();
+						let configurationFileContent = this.$fs.readText(platformData.configurationFilePath).wait();
+						let resultXml = this.mergeXml(pluginConfigurationFileContent, configurationFileContent).wait();
+						this.$fs.writeFile(platformData.configurationFilePath, resultXml).wait();	
+					}
+					
+					if(this.$fs.exists(pluginPlatformsFolderPath).wait()) {
+						shelljs.rm("-rf", pluginPlatformsFolderPath);			
+					}						
+					
+					// TODO: Add libraries
+					
+					// Show message
+					this.$logger.out(`Successfully prepared plugin ${pluginData.name} for ${platform}.`);
+				}
+			};
+			
+			this.executeForAllInstalledPlatforms(action);
+		}).future<void>()();
+	}
+	
+	public getAllInstalledPlugins(): IFuture<IPluginData[]> {
+		return (() => {
+			let nodeModules = this.$fs.readDirectory(this.nodeModulesPath).wait();
+			let plugins: IPluginData[] = [];
+			_.each(nodeModules, nodeModuleName => {
+				let nodeModuleData = this.getNodeModuleData(nodeModuleName);
+				if(nodeModuleData.isPlugin) {
+					plugins.push(this.convertToPluginData(nodeModuleData));
+				}
+			});
+			
+			return plugins;
+		}).future<IPluginData[]>()();
+	}
+	
+	private executeNpmCommand(npmCommandName: string, npmCommandArguments: string): IFuture<string> {
+		return (() => {
+			let command = this.composeNpmCommand(npmCommandName, npmCommandArguments);
+			let result = this.$childProcess.exec(command, { cwd: this.$projectData.projectDir }).wait();
+			return this.parseNpmCommandResult(result);
+		}).future<string>()();
+	}
+	
+	private composeNpmCommand(npmCommandName: string, npmCommandArguments: string): string {
+		let command = ` npm ${npmCommandName} ${npmCommandArguments} --save `;
+		if(this.$options.production) {
+			command += " --production ";
+		}
+		
+		return command;
+	}
+	
+	private parseNpmCommandResult(npmCommandResult: string): string { // The npmCommandResult is in the following format: [<name>@<version node_modules/<name>]
+		return npmCommandResult.split("@")[0]; // returns plugin name
+	}
+	
+	private executeForAllInstalledPlatforms(action: (pluginDestinationPath: string, platform: string, platformData: IPlatformData) => void): void {
+		let availablePlatforms = _.keys(this.$platformsData.availablePlatforms);
+		_.each(availablePlatforms, platform => {
+			let isPlatformInstalled = this.$fs.exists(path.join(this.$projectData.platformsDir, platform.toLowerCase())).wait();
```

Comment:
Actually I wanted to reuse it. The problem is that `pluginsService` is dependency of `platformsService` so here I am not able to inject `pluginsService` :smile: